File: frontend/src/types/space.ts
export interface SpaceItemCounts {
  calls: number;
  chats: number;
  statuses: number;
  media: number;
}

export interface Space {
  id: string;
  name: string;
  description: string | null;
  color: string;
  created_at: string;
  updated_at: string;
  item_counts: SpaceItemCounts;
}

export interface SpaceListResponse {
  spaces: Space[];
  total: number;
}

export interface SpaceCreate {
  name: string;
  description?: string;
  color?: string;
}

export interface SpaceUpdate {
  name?: string;
  description?: string | null;
  color?: string;
}

File: backend/app/schemas/space.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class SpaceCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = None
    color: str = Field(default="#7c3aed", pattern=r"^#[0-9a-fA-F]{6}$")


class SpaceUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    color: str | None = Field(default=None, pattern=r"^#[0-9a-fA-F]{6}$")


class SpaceItemCounts(BaseModel):
    calls: int = 0
    chats: int = 0
    statuses: int = 0
    media: int = 0


class SpaceResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    color: str
    created_at: datetime
    updated_at: datetime
    item_counts: SpaceItemCounts = SpaceItemCounts()

    model_config = {"from_attributes": True}


class SpaceListResponse(BaseModel):
    spaces: list[SpaceResponse]
    total: int
